### DISCUSSION TITLE
Memory leak

### DISCUSSION THREAD

**Discussion by fa1l1l** [Whisky-App/Whisky#1185] [Q&A]
I use MBA M1 8/256gb, macOS 15, Wine v9.0, Whisky 2.3.3

I play the Kenshi game (dx11, without dxvk), after a certain time, when the game locations are loaded as the characters move, the RAM and SWAP are filled and there is a sharp decrease in performance and if it continues - a complete freeze. When I used Parallels, there was no such problem, but the performance was not satisfied, in Whisky it is better, but with a "memory leak". The resources of the game are not trivially unloaded from memory until I restart it. And so on in a circle. I ask for help in the form of some startup parameters that in theory can solve this problem, or a link to a topic in which there is already a solution to such an issue, because I did not find

**Comment by Gcenx** [comment]
Whisky doesn’t use wine-9.0 but wine-7.7 so if you’ve forced wine-9.0 this wouldn’t be a Whisky problem.

According to Apple D3DMetal requires at minimum 16GB of ram.

**Comment by fa1l1l** [comment]
> Whisky doesn’t use wine-9.0 but wine-7.7 so if you’ve forced wine-9.0 this wouldn’t be a Whisky problem.
> 
> 
> 
> According to Apple D3DMetal requires at minimum 16GB of ram.

Yes, I saw yesterday that it’s still wine-7.7, it’s just that for some reason I have version 9.0 installed separately. 

And why exactly 16Gb? After all, initially the game starts with my graphics settings and at the same time does not even climb into Swap. But then it just loads new locations, clogs the memory and doesn’t unload old data. On ARM win11 through parallels, there was no such thing, the data was downloaded, but there is a problem with the small amount of memory initially available for the game because of Windows. I tried to use the ulimit -v 6144000 command to limit the use of RAM, but everything was in vain. 

Then I’ll try the option with the manual method through GPTool

(sorry of my english, a little but of google translate)
